tests/apollo/test_skvbc_preexecution.py

```diff
@@ -82,44 +87,72 @@ async def run_concurrent_pre_execution_requests(self, skvbc, clients, num_of_req
         """
-        Use a random client to launch one pre-process request in time and ensure that created block are as expected.
+        Use a random client to launch one pre-process request in time and ensure that created blocks are as expected.
         """
         bft_network.start_all_replicas()
         skvbc = kvbc.SimpleKVBCProtocol(bft_network)
 
-        for i in range(100):
+        for i in range(NUM_OF_SEQ_WRITES):
             client = bft_network.random_client()
             await self.send_single_write_with_pre_execution(skvbc, client)
 
+    @unittest.skip("unstable")
     @with_trio
     @with_bft_network(start_replica_cmd)
     async def test_concurrent_pre_process_requests(self, bft_network):
         """
-        Launch concurrent requests from different clients in parallel. Ensure that created block are as expected.
+        Launch concurrent requests from different clients in parallel. Ensure that created blocks are as expected.
         """
         bft_network.start_all_replicas()
         skvbc = kvbc.SimpleKVBCProtocol(bft_network)
 
-        clients = bft_network.clients.values()
+        clients = bft_network.random_clients(MAX_CONCURRENCY)
         num_of_requests = len(clients)
         sent_count = await self.run_concurrent_pre_execution_requests(skvbc, clients, num_of_requests)
         self.assertTrue(sent_count >= num_of_requests)
 
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    async def test_long_time_executed_pre_process_request(self, bft_network):
+        """
+        Launch pre-process request with a long-time execution and ensure that created blocks are as expected
+        and no view-change was triggered.
+        """
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        client = bft_network.random_client()
+        client.config = client.config._replace(req_timeout_milli=LONG_REQ_TIMEOUT_MILLI, retry_timeout_milli=1000)
+        await self.send_single_write_with_pre_execution(skvbc, client, long_exec=True)
+
+        clients = bft_network.clients.values()
+        with trio.move_on_after(seconds=1):
+            await self.send_indefinite_pre_execution_requests(skvbc, clients)
+
+        initial_primary = 0
+        await bft_network.wait_for_view(replica_id=initial_primary,
+                                        expected=lambda v: v == initial_primary,
+                                        err_msg="Make sure the view did not change.")
+
     @with_trio
     @with_bft_network(start_replica_cmd)
     async def test_view_change(self, bft_network):
         """
-        Crash the primary replica and verify that the system triggers a view change and moves to a new view
+        Crash the primary replica and verify that the system triggers a view change and moves to a new view.
         """
         bft_network.start_all_replicas()
         skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        await trio.sleep(5)
```

Comment:
Why is this sleep necessary, right after starting the replicas?

Reply:
I realized that the thread pool is not always ready immediately after replicas startup and the first request gets a timeout. It's not an issue in the real environment but causes tests to be unstable.